packages/sanity/src/core/form/store/formState.ts

```diff
@@ -603,26 +603,13 @@ function prepareObjectInputState<T>(
 
       const fieldsetMembers = fieldSet.fields.flatMap(
         (field): (FieldMember | FieldError | HiddenField)[] => {
-          const hidden = resolveConditionalProperty(field.type.hidden, conditionalPropertyContext)
-
-          if (fieldsetHidden || hidden) {
-            return [
-              {
-                kind: 'hidden',
-                key: `field-${field.name}`,
-                name: field.name,
-                index: index,
-              },
-            ]
-          }
```

Comment:
Removing this fixes the issue, since prepareFieldMember does the conditional hidden check (passing the correct context data) and returns a `HiddenField` if the conditional hidden callback returns true.